src/commands/fishy/fishy.ts

```diff
@@ -96,22 +97,22 @@ export default class FishyCommand extends Command {
       if (isDmChannel(message) || isInChannel(message, allChannels.fishy) || !isHomeGuild(message)) {
         if (canFish === true) {
           const total = fish.reduce((acc, cur) => acc + cur.weight, 0);
-          const threshold = Math.random() * total;
+          const threshold = crypto.randomInt(0, total);
 
           let sum = 0;
           const group = fish.find((group) => {
             sum += group.weight;
             return sum >= threshold;
           });
 
-          const index = Math.floor(Math.random() * group.puns.length);
+          const index = crypto.randomInt(0, group.puns.length - 1);
```

Comment:
*max* is non-inclusive, so you shouldn't subtract one here
you can omit *min* here too